nsxt/policy_utils_test.go

```diff
@@ -0,0 +1,101 @@
+/* Copyright © 2024 Broadcom, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type policyPathTest struct {
+	path    string
+	parents []string
+}
+
+func TestParseStandardPolicyPath(t *testing.T) {
+
+	testData := []policyPathTest{
+		{
+			path:    "/infra/tier-1s/mygw1",
+			parents: []string{"mygw1"},
+		},
+		{
+			path:    "/global-infra/tier-0s/mygw1",
+			parents: []string{"mygw1"},
+		},
+		{
+			path:    "/orgs/myorg/projects/myproj/infra/tier-1s/mygw1",
+			parents: []string{"myorg", "myproj", "mygw1"},
+		},
+		{
+			path:    "/orgs/myorg/projects/myproj/vpcs/myvpc/tier-1s/mygw1",
+			parents: []string{"myorg", "myproj", "myvpc", "mygw1"},
+		},
+		{
+			path:    "/orgs/myorg/projects/myproj/infra/domains/d1/groups/g1",
+			parents: []string{"myorg", "myproj", "d1", "g1"},
+		},
+		{
+			path:    "/global-infra/tier-1s/t15/ipsec-vpn-services/default/sessions/xxx-yyy-xxx",
+			parents: []string{"t15", "default", "xxx-yyy-xxx"},
+		},
+		{
+			path:    "/infra/evpn-tenant-configs/{config-id}",
+			parents: []string{"{config-id}"},
+		},
+		{
+			path:    "/infra/tier-0s/{tier-0-id}/locale-services/{locale-service-id}/service-interfaces/{interface-id}",
```

Comment:
silly question: are the '{ }' brackets useful for testing something in particular, or is it just a copy & paste thing?

Reply:
I thought we might test special characters as well :)